In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('data/income_evaluation.csv')
df.head()

age          workclass   fnlwgt   education   education-num  \
0   39          State-gov    77516   Bachelors              13   
1   50   Self-emp-not-inc    83311   Bachelors              13   
2   38            Private   215646     HS-grad               9   
3   53            Private   234721        11th               7   
4   28            Private   338409   Bachelors              13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

    capital-gain   capital-loss   hours-per-week  native-country  income  
0           2174              0               40   United-States   <=50K  
1              0              0               13   United-States   <=50K  
2              0              0               40   United-States   <=50K  
3              0              0               40   United-States   <=50K  
4              0              0               40            Cuba   <=50K

In [3]:
df.shape

(32561, 15)

In [4]:
df = df.sample(frac=1)

In [5]:
df

age          workclass   fnlwgt      education   education-num  \
1755    25            Private   169323      Bachelors              13   
29885   46            Private   336984        HS-grad               9   
2645    46            Private   138626   Some-college              10   
18893   53          Local-gov   197054        Masters              14   
9468    40          State-gov   199381        Masters              14   
6071    57   Self-emp-not-inc   286836           10th               6   
17713   21            Private   189013   Some-college              10   
15468   29            Private    37933           12th               8   
22801   26            Private   149734        HS-grad               9   
11061   43       Self-emp-inc   175715        HS-grad               9   
1614    26            Private   161007      Bachelors              13   
10129   37            Private   167851        HS-grad               9   
21154   22            Private   219042           10th               6   
31706   29            Private    50295   Some-college              10   
21380   60            Private   186446        7th-8th               4   
16798   51                  ?   146325        HS-grad               9   
29356   52   Self-emp-not-inc   138611        HS-grad               9   
13425   75   Self-emp-not-inc    92792        HS-grad               9   
19655   32            Private   207668      Bachelors              13   
5406    90            Private    51744        Masters              14   
5505    29            Private   632593        HS-grad               9   
24850   48            Private   107231    Prof-school              15   
21058   44   Self-emp-not-inc   172479        HS-grad               9   
6223    27            Private    72208   Some-college              10   
11371   63            Private   203783        HS-grad               9   
21087   70            Private   280307      Bachelors              13   
25102   51            Private    55500           12th               8   
6200    37   Self-emp-not-inc   415847      Bachelors              13   
843     62          Local-gov   291175      Bachelors              13   
23173   19            Private   254247           12th               8   
...    ...                ...      ...            ...             ...   
1517    59            Private   284834      Bachelors              13   
18118   39            Private   280570      Bachelors              13   
28845   31            Private   272856        HS-grad               9   
31298   29        Federal-gov   242147        HS-grad               9   
23143   26            Private   456618        HS-grad               9   
2261    27            Private   408623      Bachelors              13   
6441    28            Private   338376      Bachelors              13   
21784   32            Private   248584        HS-grad               9   
3661    57            Private   145189   Some-college              10   
24586   33            Private   189265      Bachelors              13   
4850    60   Self-emp-not-inc   376973        HS-grad               9   
1285    46            Private   188861   Some-college              10   
22829   61        Federal-gov   151369    Prof-school              15   
30965   54          Local-gov   127943        HS-grad               9   
29441   61            Private   316359   Some-college              10   
14845   20            Private    60639   Some-college              10   
208     58       Self-emp-inc   210563        HS-grad               9   
8007    44                  ?   174147   Some-college              10   
15493   67       Self-emp-inc    81413        HS-grad               9   
6       49            Private   160187            9th               5   
26081   40                  ?   165309        7th-8th               4   
2828    54   Self-emp-not-inc   175804      Bachelors              13   
17008   38          Local-gov   236687   Some-college            

In [6]:
len(df) * 0.8

26048.800000000003

In [7]:
train = df[:26048]

In [8]:
test = df[26048:]

In [9]:
train.head()

age   workclass   fnlwgt      education   education-num  \
1755    25     Private   169323      Bachelors              13   
29885   46     Private   336984        HS-grad               9   
2645    46     Private   138626   Some-college              10   
18893   53   Local-gov   197054        Masters              14   
9468    40   State-gov   199381        Masters              14   

            marital-status          occupation  relationship    race      sex  \
1755    Married-civ-spouse     Exec-managerial     Own-child   White   Female   
29885        Never-married   Handlers-cleaners     Own-child   White     Male   
2645    Married-civ-spouse        Craft-repair       Husband   White     Male   
18893   Married-civ-spouse      Prof-specialty       Husband   White     Male   
9468    Married-civ-spouse      Prof-specialty          Wife   White   Female   

        capital-gain   capital-loss   hours-per-week  native-country  income  
1755               0              0               40   United-States   <=50K  
29885              0              0               17   United-States   <=50K  
2645               0              0               40   United-States   <=50K  
18893              0              0               40   United-States    >50K  
9468           15024              0               37   United-States    >50K

In [10]:
train.columns

Index(['age', ' workclass', ' fnlwgt', ' education', ' education-num',
       ' marital-status', ' occupation', ' relationship', ' race', ' sex',
       ' capital-gain', ' capital-loss', ' hours-per-week', ' native-country',
       ' income'],
      dtype='object')

In [11]:
train.columns = train.columns.str.lstrip()

In [12]:
test.columns = test.columns.str.lstrip()

In [13]:
X_train = train.drop('income', axis=1)

In [14]:
X_train.head()

age   workclass  fnlwgt      education  education-num  \
1755    25     Private  169323      Bachelors             13   
29885   46     Private  336984        HS-grad              9   
2645    46     Private  138626   Some-college             10   
18893   53   Local-gov  197054        Masters             14   
9468    40   State-gov  199381        Masters             14   

            marital-status          occupation relationship    race      sex  \
1755    Married-civ-spouse     Exec-managerial    Own-child   White   Female   
29885        Never-married   Handlers-cleaners    Own-child   White     Male   
2645    Married-civ-spouse        Craft-repair      Husband   White     Male   
18893   Married-civ-spouse      Prof-specialty      Husband   White     Male   
9468    Married-civ-spouse      Prof-specialty         Wife   White   Female   

       capital-gain  capital-loss  hours-per-week  native-country  
1755              0             0              40   United-States  
29885             0             0              17   United-States  
2645              0             0              40   United-States  
18893             0             0              40   United-States  
9468          15024             0              37   United-States

In [15]:
y_train = train.income

In [16]:
y_train.head()

1755      <=50K
29885     <=50K
2645      <=50K
18893      >50K
9468       >50K
Name: income, dtype: object

In [17]:
X_test = test.drop('income', axis='columns')

In [18]:
y_test = test.income

# Using sklearn's train_test_split

In [19]:
d = pd.read_csv('data/income_evaluation.csv')
d.head()

age          workclass   fnlwgt   education   education-num  \
0   39          State-gov    77516   Bachelors              13   
1   50   Self-emp-not-inc    83311   Bachelors              13   
2   38            Private   215646     HS-grad               9   
3   53            Private   234721        11th               7   
4   28            Private   338409   Bachelors              13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

    capital-gain   capital-loss   hours-per-week  native-country  income  
0           2174              0               40   United-States   <=50K  
1              0              0               13   United-States   <=50K  
2              0              0               40   United-States   <=50K  
3              0              0               40   United-States   <=50K  
4              0              0               40            Cuba   <=50K

In [20]:
from sklearn.model_selection import train_test_split

In [21]:
X = d.drop(' income', axis=1)
y = d[' income']

In [22]:
X1, X2, y1, y2 = train_test_split(X, y, test_size=0.2, random_state=0, stratify=y)

In [23]:
y1.value_counts()

 <=50K    19775
 >50K      6273
Name:  income, dtype: int64

In [24]:
y2.value_counts()

 <=50K    4945
 >50K     1568
Name:  income, dtype: int64

In [25]:
y.value_counts()

 <=50K    24720
 >50K      7841
Name:  income, dtype: int64

24720 is the total number of "<=50K" data samples. <br>These will be split in the ratio of 80:20

In the 'training y', i.e. y1, we'll have 24720 * 0.8 = 19776 "<=50K" samples

In [26]:
24720*0.8

19776.0

In the 'testing y', i.e. y2, we'll have 24720 * 0.2 = 4944 "<=50K" samples

In [27]:
24720*0.2

4944.0

7841 is the total number of ">50K" data samples. <br>These will be split in the ratio of 80:20

In the 'testing y', i.e. y2, we'll have 7841 * 0.2 = 1568 ">50K" samples (rounding off)

In [28]:
7841*0.2

1568.2

In the 'training y', i.e. y1, we'll have 7841 * 0.8 = 6273 ">50K" samples (rounding off)

In [29]:
7841*0.8

6272.8

In [30]:
X1.head()

age  workclass   fnlwgt      education   education-num  \
29101   19          ?   140399   Some-college              10   
10606   50    Private   158284   Some-college              10   
9414    62    Private   183735   Some-college              10   
1747    20    Private   154781        HS-grad               9   
10677   25    Private   356344      Bachelors              13   

            marital-status          occupation     relationship    race  \
29101        Never-married                   ?   Other-relative   White   
10606   Married-civ-spouse   Machine-op-inspct          Husband   White   
9414    Married-civ-spouse     Exec-managerial          Husband   White   
1747         Never-married        Adm-clerical    Not-in-family   White   
10677        Never-married        Adm-clerical        Own-child   White   

           sex   capital-gain   capital-loss   hours-per-week  native-country  
29101   Female              0              0               30   United-States  
10606     Male              0              0               40   United-States  
9414      Male              0              0               40   United-States  
1747    Female              0              0               40   United-States  
10677   Female              0              0               40   United-States